ocs_ci/utility/assisted_installer.py

```diff
@@ -0,0 +1,517 @@
+        }
+        resp = self.get_request(
+            f"clusters/{cluster_id}/downloads/files", params=params, json=False
+        )
+        filenames = re.findall(
+            "filename=(.+)", resp.headers.get("content-disposition", "")
+        )
+        if filenames:
+            filename = filenames[0].strip("\"'")
+        else:
+            filename = file_name
+        dest_dir = os.path.expanduser(dest_dir)
+        os.makedirs(dest_dir, exist_ok=True)
+        with open(os.path.join(dest_dir, filename), "wb") as fd:
+            fd.write(resp.content)
+
+    def delete_cluster(self, cluster_id):
+        """
+        Delete cluster
+
+        Args:
+            cluster_id (str): cluster ID
+        """
+        resp = self.delete_request(f"clusters/{cluster_id}", json=False)
+        return resp.ok
+
+    def get_infra_envs(self):
+        """
+        Get list of infra_envs.
+
+        Returns:
+            dict: the infra_envs configuration
+        """
+        return self.get_request("infra-envs")
+
+    def get_infra_env(self, infra_env_id):
+        """
+        Get information about one infra_env.
+
+        Args:
+            infra_env_id (str): Infra environment ID
+
+        Returns:
+            dict: the infra_env configuration
+        """
+        return self.get_request(f"infra-envs/{infra_env_id}")
+
+    def create_infra_env(self, data):
+        """
+        Create (register) new Infrastructure Environment in Assisted Installer console
+
+        Args:
+            data (dict): Infrastructure environment configuration based on
+                https://api.openshift.com/?urls.primaryName=assisted-service%20service#/installer/RegisterInfraEnv
+
+        """
+        return self.post_request("infra-envs", data)
+
+    def get_infra_env_hosts(self, infra_env_id):
+        """
+        Get information about hosts from infra_env.
+
+        Args:
+            infra_env_id (str): Infra environment ID
+
+        Returns:
+            dict: the hosts configuration
+        """
+        return self.get_request(f"infra-envs/{infra_env_id}/hosts")
+
+    def get_infra_env_host(self, infra_env_id, host_id):
+        """
+        Get information about host from infra_env.
+
+        Args:
+            infra_env_id (str): Infra environment ID
+            host_id (str): host ID
+
+        Returns:
+            dict: the host configuration
+        """
+        return self.get_request(f"infra-envs/{infra_env_id}/hosts/{host_id}")
+
+    def update_infra_env_host(self, infra_env_id, host_id, update_data):
+        """
+        Update host configuration
+
+        Args:
+            infra_env_id (str): Infra environment ID
+            host_id (str): Host ID
+            update_data (dict): the data to be updated
+        """
+        data = update_data
+        return self.patch_request(
+            f"infra-envs/{infra_env_id}/hosts/{host_id}", data=data
+        )
+
+    def get_discovery_iso_url(self, infra_env_id):
```

Comment:
for my understanding, can we use discovery to setup VM instead of boot iso?

Reply:
I'm not sure, what you meant?